android/src/com/mapswithme/maps/widget/placepage/PlacePageView.java

```diff
@@ -1154,6 +1213,42 @@ public void onClick(View v)
       case R.id.ll__place_website:
         Utils.openUrl(getContext(), mTvWebsite.getText().toString());
         break;
+      case R.id.ll__place_facebook:
+        if (mMapObject == null)
+        {
+          LOGGER.e(TAG, "Cannot follow facebook url, mMapObject is null!");
+          break;
+        }
+        final String facebookPage = mMapObject.getMetadata(Metadata.MetadataType.FMD_FACEBOOK_PAGE);
```

Comment:
Other fields are fetched from text editors, not from the map object. What is wrong with that?

Reply:
Text editor could contains either `@<account-name>` string or full url. You gonna need to check it's format. With mMapObject it's easier to just prepend value with `https://<mobile-domain>/` to get full URL.